daemon/api.go

```diff
@@ -823,10 +850,19 @@ func sideloadSnap(c *Command, r *http.Request, user *auth.UserState) Response {
 
 	var flags snapstate.Flags
 
-	if len(form.Value["devmode"]) > 0 && form.Value["devmode"][0] == "true" {
+	if isFormValueTrue(form.Value["devmode"]) {
 		flags |= snapstate.DevMode
 	}
+	if isFormValueTrue(form.Value["confined"]) {
+		if flags.DevMode() {
```

Comment:
Same logic twice in slightly different shapes. Might instead be on both:
```
err := checkFlags(flags)
```
Would also allow the errors to be inlined.